drizzlepac/hlautils/astrometric_utils.py

```diff
@@ -700,38 +701,49 @@ def extract_sources(img, dqmask=None, fwhm=3.0, kernel=None, photmode=None,
             # Pick out brightest source only
             if src_table is None and seg_table:
                 # Initialize final master source list catalog
+                log.debug("Defining initial src_table based on: {}".format(seg_table.colnames))
                 src_table = Table(names=seg_table.colnames,
                                   dtype=[dt[1] for dt in seg_table.dtype.descr])
 
             if seg_table:
                 # This logic will eliminate saturated sources, where the max pixel value is not
                 # the center of the PSF (saturated and streaked along the Y axis)
                 max_row = np.where(seg_table['peak'] == seg_table['peak'].max())[0][0]
-                peak_posx = int(seg_table[max_row]['xcentroid'] + 0.5)
-                peak_posy = int(seg_table[max_row]['ycentroid'] + 0.5)
-                delta = (source_box - 1) // 2
-                min_x = peak_posx - delta if peak_posx - delta > 0 else 0
-                max_x = peak_posx + delta + 1 if peak_posx + delta + 1 < seg_slice[1].stop else seg_slice[1].stop
-                min_y = peak_posy - delta if peak_posy - delta > 0 else 0
-                max_y = peak_posy + delta + 1 if peak_posy + delta + 1 < seg_slice[0].stop else seg_slice[0].stop
-                peak_region = detection_img[min_y:max_y, min_x:max_x]
-
-                if np.isclose(peak_region, seg_table['peak'].max()).any():
-                    # Add row for detected source to master catalog
-                    # apply offset to slice to convert positions into full-frame coordinates
-                    seg_table['xcentroid'] += seg_xoffset
-                    seg_table['ycentroid'] += seg_yoffset
-                    src_table.add_row(seg_table[max_row])
+
+                # Add logic to remove sources which have more than 8 pixels
+                # within 10% of the max value in the source segment, a situation
+                # which would indicate bleeding from a saturated source
+                if (detection_img > detection_img.max() * 0.9).sum() > 3:
```

Comment:
Which is incorrect here - the "3" in the code or the "8" in the comment?  Or is this doing something I do not understand?

Reply:
This comment was corrected in 6b78487 to report on the value of '3' being used as a radius of 0.5 pixels area where all pixels are 'the same' to indicate a saturated source.